File: backend/app/schemas/third_party.py
"""Pydantic schemas for third parties."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class ThirdPartyOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    claim_id: UUID
    kind: str
    full_name: str
    document_id: str | None
    contact_phone: str | None
    vehicle_plate: str | None
    vehicle_info: str | None
    statement: str | None
    tenant_id: UUID
    created_at: datetime
    updated_at: datetime | None


class ThirdPartyCreate(BaseModel):
    kind: str = Field(min_length=1, max_length=50)
    full_name: str = Field(min_length=3, max_length=200)
    document_id: str | None = None
    contact_phone: str | None = None
    vehicle_plate: str | None = None
    vehicle_info: str | None = None
    statement: str | None = None


class ThirdPartyUpdate(BaseModel):
    kind: str | None = None
    full_name: str | None = Field(default=None, min_length=3, max_length=200)
    document_id: str | None = None
    contact_phone: str | None = None
    vehicle_plate: str | None = None
    vehicle_info: str | None = None
    statement: str | None = None


class ThirdPartyListResponse(BaseModel):
    items: list[ThirdPartyOut]
    total: int
    page: int
    limit: int

File: frontend/types/third-party.ts
export interface ThirdParty {
  id: string
  claim_id: string
  kind: string
  full_name: string
  document_id: string | null
  contact_phone: string | null
  vehicle_plate: string | null
  vehicle_info: string | null
  statement: string | null
  tenant_id: string
  created_at: string
  updated_at: string | null
}

export interface ThirdPartyListResponse {
  items: ThirdParty[]
  total: number
  page: number
  limit: number
}

export interface ThirdPartyCreatePayload {
  kind: string
  full_name: string
  document_id?: string
  contact_phone?: string
  vehicle_plate?: string
  vehicle_info?: string
  statement?: string
}

export interface ThirdPartyUpdatePayload {
  kind?: string
  full_name?: string
  document_id?: string
  contact_phone?: string
  vehicle_plate?: string
  vehicle_info?: string
  statement?: string
}
